src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowMarkdownCommand.cs

```diff
@@ -28,46 +30,119 @@ public class ShowMarkdownCommand : PSCmdlet
-            }
+            _powerShell = System.Management.Automation.PowerShell.Create();
         }
 
         /// <summary>
         /// Override ProcessRecord.
         /// </summary>
         protected override void ProcessRecord()
         {
-            object inpObj = InputObject.BaseObject;
+            switch (ParameterSetName)
+            {
+                case "InputObject":
+                    if (InputObject.BaseObject is MarkdownInfo)
+                    {
+                        ProcessMarkdownInfo(InputObject.BaseObject);
+                    }
+                    else
+                    {
+                        ConvertFromMarkdown("InputObject", InputObject.BaseObject);
+                    }
+
+                    break;
+
+                case "Path":
+                case "LiteralPath":
+                    ConvertFromMarkdown(ParameterSetName, Path);
+                    break;
+
+                default:
+                    Debug.Assert(false, string.Format(CultureInfo.InvariantCulture, "Invalid parameter set name: {0}", ParameterSetName));
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Process markdown as path.
+        /// </summary>
+        /// <param name="parameter">Name of parameter to pass to `ConvertFrom-Markdown`.</param>
+        /// <param name="input">Value of parameter.</param>
+        private void ConvertFromMarkdown(string parameter, object input)
+        {
+            _powerShell.AddCommand("Microsoft.PowerShell.Utility\\ConvertFrom-Markdown").AddParameter(parameter, input);
+            if (!UseBrowser)
+            {
+                _powerShell.AddParameter("AsVT100EncodedString");
+            }
+
+            Collection<PSObject> output = _powerShell.Invoke();
+
+            if (_powerShell.HadErrors)
+            {
+                foreach (ErrorRecord errorRecord in _powerShell.Streams.Error)
+                {
+                    WriteError(errorRecord);
+                }
+            }
+
+            foreach (PSObject result in output)
+            {
+                ProcessMarkdownInfo(result.BaseObject);
+            }
+        }
 
-            if (inpObj is MarkdownInfo markdownInfo)
+        /// <summary>
+        /// Process markdown as input objects.
+        /// </summary>
+        /// <param name="inputObject">Markdown object to process.</param>
+        private void ProcessMarkdownInfo(object inputObject)
+        {
+            if (inputObject is MarkdownInfo markdownInfo)
             {
                 if (UseBrowser)
                 {
                     var html = markdownInfo.Html;
 
                     if (!string.IsNullOrEmpty(html))
                     {
-                        string tmpFilePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + ".html");
+                        string tmpFilePath = System.IO.Path.Combine(System.IO.Path.GetTempPath(), Guid.NewGuid().ToString() + ".html");
```

Comment:
I believe we can remove `System.IO`.

Reply:
The compiler complained that `Path` type here was confused with the `Path` parameter.